### PR TITLE
Improvements on boutiques input cache cleaner.

### PR DESCRIPTION
**Pull request by prioux**
The cleaning now only occurs if the input file was not already
synchronized at the moment we enter setup().

### CODE REVIEW

File: BrainPortal/lib/boutiques_input_cache_cleaner.rb [hunk 1]
```diff
@@ -36,44 +36,72 @@
...
+# In the example above, the cached content of any userfile(s) selected for the
+# file inputs named 'my_input1' or 'my_input2' will be deleted after the task
+# completes properly. This happens unless CBRAIN notices that another task
+# uses that input too. The module tries to handle conflict between tasks that
+# share same file based on the SyncStatus timestamps.
 module BoutiquesInputCacheCleaner
 
   def setup #:nodoc:
+
+    # BEFORE code: identify userfiles that are not synced at all
```
Comment (MontrealSergiy) [inline BrainPortal/lib/boutiques_input_cache_cleaner.rb:48]: `# BEFORE standard setup:  identify userfiles that are not synced at all `

would be clearer description. Also the class doc could mention that feature only work for nonsync files.

Comment (prioux) [reply BrainPortal/lib/boutiques_input_cache_cleaner.rb]: I improved the documentation.

**Comment by MontrealSergiy** [comment]
On a few tests seems work as intended - sync files remain sync, async remain async

**Review by MontrealSergiy — APPROVED** [review]
The idea to reduce chances of erroneous erase seems valid, chances of erroneous erase are low

**Pull request merged**